ekncontent/ekncontent/eknc-domain.c

```diff
@@ -924,6 +924,28 @@ on_xapian_query_response (GObject *source,
     }
 }
 
+void
+on_xapian_test_response (GObject *source,
+                         GAsyncResult *result,
+                         gpointer user_data)
+{
+  EkncXapianBridge *bridge = EKNC_XAPIAN_BRIDGE (source);
+  g_autoptr(GTask) task = user_data;
+  EkncDomain *domain = g_task_get_source_object (task);
+  GCancellable *cancellable = g_task_get_cancellable (task);
+
+  eknc_xapian_bridge_test_finish (bridge, result);
+
+  EkncQueryObject *query = g_task_get_task_data (task);
+
+  QueryState *state = g_slice_new0 (QueryState);
```

Comment:
I would want to refactor this code into a function that gets called by both here and the code path when the feature test is not needed below.